### ISSUE TITLE
Examples failing with NodeJS 18 - use of require

### ISSUE THREAD

**Issue by hardillb** [andreialecu/dmarc-report-parser#1]
Hi

I'm trying to run your sample code and it's throwing errors about using `require` from a ESM module

version 0.1.4 from npm

```
import { parseDmarcReportsFromXml } from "dmarc-report-parser";
import { readFile} from 'node:fs/promises'

const file = await readFile('test/google.com!example.com!1705622400!1705708799.xml')

const xmlFiles = [file]; // An array of XML strings or Buffers

try {
  const result = await parseDmarcReportsFromXml(xmlFiles);
  console.log(result);
} catch (error) {
  console.error("Failed to parse DMARC report:", error.message);
}
```

Error:
```
$ node index.mjs 
/opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/dmarc-feedback-schema.js:7
const camelcase_keys_1 = __importDefault(require("camelcase-keys"));
                                         ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /opt/share/playing/dmarc/node_modules/camelcase-keys/index.js from /opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/dmarc-feedback-schema.js not supported.
Instead change the require of index.js in /opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/dmarc-feedback-schema.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/dmarc-feedback-schema.js:7:42)
    at Object.<anonymous> (/opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/parseDmarcReportsFromXml.js:5:33)
    at Object.<anonymous> (/opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/parseDmarcReportFromEmail.js:10:36)
    at Object.<anonymous> (/opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/index.js:17:14) {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v18.19.0
```

TypeScript isn't really my thing, but I assume this is an artefact of the compile?

**Comment by andreialecu** [comment]
It appears you're using an `index.mjs` file, but this is built and distributed as commonjs.

Unfortunately, .mjs are not really my thing either. As a quick workaround, you could try an `index.js` with `require`s instead of `import`s.

**Comment by hardillb** [comment]
I get the same error

```
const parseDmarcReportsFromXml = require("dmarc-report-parser")
const fs =  require('node:fs/promises')

const parse = async function () {
  const file = await fs.readFile('test/google.com!example.com!1705622400!1705708799.xml')

  const xmlFiles = [file]; // An array of XML strings or Buffers

  try {
    const result = await parseDmarcReportsFromXml(xmlFiles)
    console.log(result)
  } catch (error) {
    console.error("Failed to parse DMARC report:", error.message)
  }
}

parse()
```

error:

```
$ node index.js 
/opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/dmarc-feedback-schema.js:7
const camelcase_keys_1 = __importDefault(require("camelcase-keys"));
                                         ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /opt/share/playing/dmarc/node_modules/camelcase-keys/index.js from /opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/dmarc-feedback-schema.js not supported.
Instead change the require of index.js in /opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/dmarc-feedback-schema.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/dmarc-feedback-schema.js:7:42)
    at Object.<anonymous> (/opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/parseDmarcReportsFromXml.js:5:33)
    at Object.<anonymous> (/opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/parseDmarcReportFromEmail.js:10:36)
    at Object.<anonymous> (/opt/share/playing/dmarc/node_modules/dmarc-report-parser/dist/index.js:17:14)
    at Object.<anonymous> (/opt/share/playing/dmarc/index.js:1:34) {
  code: 'ERR_REQUIRE_ESM'
```

It looks like `camlecase-keys` has updated to be a ESM module so you'll need to pin to a version that is commonjs or update to use the new module.

**Comment by hardillb** [comment]
ahh, that modules is from the same folk as the got http request library, they went all in on ESM and don't support commonjs stuff at all anymore

**Comment by andreialecu** [comment]
Should be fixed in 0.1.5

Thanks for the report.

**Comment by hardillb** [comment]
Thanks, looks to work now
